cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -70,10 +71,11 @@ setNullTxOutConsumedAfterTxInId :: MonadIO m => TxOutId -> ReaderT SqlBackend m
 setNullTxOutConsumedAfterTxInId txOutId = do
   update txOutId [TxOutConsumedByTxInId =. Nothing]
 
-migrateTxOut :: MonadIO m => Maybe (Trace IO Text) -> ReaderT SqlBackend m ()
+migrateTxOut :: (MonadBaseControl IO m, MonadIO m) => Maybe (Trace IO Text) -> ReaderT SqlBackend m ()
 migrateTxOut mTrace = do
   createConsumedTxOut
   migrateNextPage 0
+  insertExtraMigration PruneTxOutFlagPreviouslySet
```

Comment:
This should be inserted only used when `prune` is enabled. Otherwise the restarts for `consume` get broken. If `consume` is set once, then restarted without `prune` db-sync will see this value in the table, but not the flag, so it will crash.

Reply:
Ah my mistake I thought this only happened when `prune` was set.